force-app/main/default/classes/RecipeTriggerHandler_Test.cls

```diff
@@ -36,129 +36,182 @@ public class RecipeTriggerHandler_Test {
+        
+        recipeRecord.Servings__c = 100;
+        recipeRecord.Active_Time_Units__c = 'Minutes';
+        recipeRecord.Active_Time__c = 10; 
+        
+        test.startTest();
+        update recipeRecord;
+        test.stopTest();
+        
+        Recipe__c updatedRecipe = [SELECT ID, Complexity__c FROM Recipe__c WHERE id = : recipeRecord.Id];
+        system.assertEquals('Moderate', updatedRecipe.Complexity__c);
+        
+    }
+    @isTest public static void validateComplexitySetAsDifficultOnUpdate(){
+        Recipe__c recipeRecord = [SELECT ID, Complexity__c FROM Recipe__c WHERE Complexity__c = 'Simple' LIMIT 1];
+        system.assertEquals('Simple', recipeRecord.Complexity__c);
+        
+        recipeRecord.Active_Time_Units__c = 'Hours';
+        recipeRecord.Active_Time__c = 30;
+        
+        test.startTest();
+        update recipeRecord; 
+        test.stopTest();
+        
+        Recipe__c updatedRecipe = [SELECT ID, Complexity__c FROM Recipe__c WHERE id = : recipeRecord.Id];
+        system.assertEquals('Difficult', updatedRecipe.Complexity__c);
+        
+    }
+    
+    @isTest public static void validateTaskCreatedOnUpdate(){
+        Recipe_Usage__c recipeUsageRecord = [SELECT ID, Recipe__c, Recipe__r.Draft__c, Cookbook__c FROM Recipe_Usage__c LIMIT 1];
+        
+        Recipe__c recipeRecord = recipeUsageRecord.Recipe__r; 
+        recipeRecord.Name = 'Test';
+        recipeRecord.Description__c = 'Test';
+        recipeRecord.Servings__c = 3; 
+        recipeRecord.Active_Time_Units__c = 'Minutes';
+        recipeRecord.Active_Time__c = 3; 
+        
+        test.startTest();
+        update recipeRecord; 
+        test.stopTest();
+        
+        Task reviewTaskRecord = [SELECT ID FROM Task WHERE WhatId = : recipeUsageRecord.Cookbook__c];
+        system.assert(reviewTaskRecord.Id != NULL);
+        
+        
+    }
+    @isTest public static void validateSObjectException(){
+        Recipe__c recipeRecord = [SELECT ID FROM Recipe__c WHERE Draft__c = FALSE LIMIT 1];
+        recipeRecord.Active_Time_Units__c = 'Ten';
+        
+        List<Boolean> exceptions = new List<Boolean>();
+        test.startTest();
+        try{
+            update recipeRecord;
+            
+        }
+        catch(sObjectException e){
+            exceptions.add(true);
+        }
+        test.stopTest();
+        system.assertNotEquals(NULL, exceptions);
+        for(Boolean b : exceptions){
+            system.assertEquals(True, b);
+        }
+    }
+    @isTest public static void validateCustomException(){
+        Recipe__c recipeRecord = new Recipe__c();
+        recipeRecord.Name = 'Test Recipe';
+        
+        List<Boolean> exceptions = new List<Boolean>();
+        test.startTest();
+        try{
+            insert recipeRecord;
+        }
+        catch(Exception e){
+            if(e.getMessage().contains('you are missing a required field')){
```

Comment:
It might make more sense to have a single boolean rather than a list. If the message is different from this line though, would the test be able to tell you that? It might make sense to have an else condition here and then validate it outside of the catch block